system/src/Grav/Common/Grav.php

```diff
@@ -520,6 +350,96 @@ public function shutdown()
     }
 
     /**
+     * Magic Catch All Function
+     * Used to call closures like measureTime on the instance.
+     * Source: http://stackoverflow.com/questions/419804/closures-as-class-members
+     */
+    public function __call($method, $args)
+    {
+        $closure = $this->$method;
+        call_user_func_array($closure, $args);
+    }
+
+    /**
+     * Initialize and return a Grav instance
+     *
+     * @param  array $values
+     *
+     * @return static
+     */
+    protected static function load(array $values)
+    {
+        $container = new static($values);
+
+        $container['grav'] = $container;
+
+        $container['debugger'] = new Debugger();
+        $debugger = $container['debugger'];
+
+        // closure that measures time by wrapping a function into startTimer and stopTimer
+        // The debugger can be passed to the closure. Should be more performant
+        // then to get it from the container all time.
+        $container->measureTime = function($timerId, $timerTitle, $callback) use ($debugger) {
+          $debugger->startTimer($timerId, $timerTitle);
+          $callback($debugger);
+          $debugger->stopTimer($timerId);
+        };
+
+        $container->measureTime('_services', 'Services', function() use ($container) {
+          self::registerServices($container);
+        });
+
+        return $container;
+    }
+
+    /**
+     * Register all services
+     * Services are defined in the diMap. They can either only the class
+     * of a Service Provider or a pair of serviceKey => serviceClass that
+     * gets directly mapped into the container.
+     *
+     * @param  Container $container
+     *
+     * @return void
+     */
+    protected static function registerServices($container) {
```

Comment:
I'm not sure why these functions are static; they are only used from `Grav::load()`, which already has container class available.